### ISSUE TITLE
`withScope()` does not get executed when Sentry is disabled

### ISSUE THREAD

**Issue by TheDevMinerTV** [getsentry/sentry-java#3030]
### Integration

sentry

### Java Version

17

### Version

6.33.0

### Steps to Reproduce

1. Paste the following code into your editor with Sentry in scope
```java
import io.sentry.Sentry;

public class Repro {
  public static void main(String[] args) {
    withoutSentry();
    withSentryDisabled();
    withSentry();
  }

  private static void withoutSentry() {
    System.out.println("without sentry:");

    System.out.println("pre Sentry.withScope");
    Sentry.withScope(s -> {
      System.out.println("in Sentry.withScope"); // <- this will never run
    });
  }

  private static void withSentryDisabled() {
    System.out.println("====\nwith sentry disabled:");

    Sentry.init(o -> o.setEnabled(false));

    System.out.println("pre Sentry.withScope");
    Sentry.withScope(s -> {
      System.out.println("in Sentry.withScope"); // <- this will never run
    });
  }

  private static void withSentry() {
    System.out.println("====\nwith sentry:");

    Sentry.init(o -> o.setDsn("<SOME VALID DSN>"));

    System.out.println("pre Sentry.withScope");
    Sentry.withScope(s -> {
      System.out.println("in Sentry.withScope"); // <- this will
    });
  }
}
```
2. Add a valid DSN into the last function

### Expected Result

In all cases, the `in Sentry.withScope` messages get logged.

### Actual Result

It only logs the message when sentry is properly configured:
![image](https://github.com/getsentry/sentry-java/assets/29845135/3a9cab73-4568-403d-b3eb-98bde8f7097e)

The docs don't say anything about this behaviour.

**Comment by adinauer** [comment]
Hey @TheDevMinerTV, thanks for opening this issue. We'll discuss internally on how to solve this.

- `NoOpHub.withScope` needs to always invoke the callback
- `Hub.withScope` needs to invoke the callback also in case Sentry has been disabled

We'll have to figure out what sort of `Scope` we want to pass into the callback in case Sentry has been disabled. We could simply use `Scope(SentryOptions)` with default `SentryOptions`.

**Comment by romtsn** [comment]
Closed by #3066
